forum/qa-include/qa-page.php

```diff
@@ -384,7 +384,13 @@ function qa_output_content($qa_content)
         if (isset($qa_content['script_src'])) {
             $uniquesrc = array_unique($qa_content['script_src']); // remove any duplicates
             foreach ($uniquesrc as $script_src) {
-                $script[] = '<script src="' . qa_html($script_src) . '?v=' . QA_RESOURCE_VERSION .'"></script>';
+                $line = '<script src="' . qa_html($script_src);

```

Comment:
Czy skrypty z tej listy są krytyczne/używane w trakcie ładowania DOM-u? Jeśli nie, to można im dać `defer` - tak jak dostały choćby skrypty SyntaxHighlightera w #216.

Reply:
Nie mam pojęcia, trzeba by poszukać co tam może się znaleźć. Na pewno skrypt od recaptchy

Reply:
Kiedy ten skrypt od recaptchy jest właściwie używany? Bo ja sobie nie przypominam, żebym go kiedyś widział.

Reply:
Podczas rejestracji. Captcha była też długi czas na formularzu kontaktowym, w tym momencie nie jest, ale trzeba zakładać że może być

https://github.com/CodersCommunity/forum.pasja-informatyki.local/blob/master/forum/qa-plugin/q2a-recaptcha-v3/src/qa-recaptcha-captcha.php#L33-L41

Reply:
Widzę tam podpięcie na `DOMContentLoaded` event - atrybut `defer` będzie jak znalazł.

Reply:
To spoko, ale jeśli chcemy to dodać do wszystkich skryptów z tablicy `script_src` to trzeba się też upewnić, że wszystkie mogą być tak podpinane

Reply:
Ok, wychodzi na to, że ta tablica nie jest specjalnie używana, znalazłem tam tylko dołączanie `ckeditor.js` oraz captchy. Wyrzuciłem więc metodę `tryDeferScript()` z `qa-theme-base.php`, która była tylko po to, aby dodać defer do `ckeditor.js` i wstawiłem go po prostu na sztywno do wszystkiego co jest wrzucane z tej tablicy.

Reply:
Spodziewałem się, że metoda `tryDeferScript` dodaje `defer` do więcej niż jednego skryptu, ale jeśli okazuje się, że nie, to niepotrzebnie ją kiedyś dodałem. Może na stronie głównej, albo na którejś z podstron przez tę metodę przewija się więcej skryptów?

Reply:
jak na moje oko tu https://github.com/CodersCommunity/forum.pasja-informatyki.local/pull/227/files#diff-8004bfb8d18026d9450ce8bbc0f1235f62da13602ac0c71ae5933bb37388708aL293 jest określona tablica do których może dodać defer i to jest tylko jeden skrypt, który zostanie obsłużony przez dodanie defer do tablicy `script_src`, stąd wyrzuciłem